src/cmd/run.js

```diff
@@ -296,7 +297,8 @@ export default async function run(
     log.debug('Not installing as temporary add-on because the ' +
               'add-on was already installed');
   } else if (requiresRemote) {
-    client = await firefoxClient();
+    const port = runningFirefox.spawnargs[2];
```

Comment:
I think you can just tell Firefox to start the debugger server on a specific port. You would pass it like [`binaryArgs.push('--listen', '6005')`](https://github.com/ccarruitero/web-ext/blob/4f4b9efd5834f2603c8bb8c4d9d399652567b206/src/cmd/run.js#L425) (see [fx-runner](https://github.com/mozilla-jetpack/node-fx-runner)).